kafka/src/main/java/io/specmesh/kafka/provision/schema/SchemaReaders.java

```
@@ -59,38 +59,32 @@ private SchemaReaders() {}
 
     public static final class FileSystemSchemaReader {
 
-        public Collection<ParsedSchema> readLocal(final String filePath) {
+        public Collection<ParsedSchema> readLocal(final Path filePath) {
             try {
-                final var schemaContent = Files.readString(Paths.get(filePath));
+                final var schemaContent = Files.readString(filePath);
                 final var results = new ArrayList<ParsedSchema>();
 
-                if (filePath.endsWith(".avsc")) {
+                final String filename =
+                        Optional.ofNullable(filePath.getFileName())
+                                .map(Objects::toString)
+                                .orElse("");
+                if (filename.endsWith(".avsc")) {
                     final var refs = resolveReferencesFor(filePath, schemaContent);
                     results.add(
                             new AvroSchema(
                                     schemaContent, refs.references, refs.resolvedReferences, -1));
-                }
-                if (filePath.endsWith(".yml")) {
+                } else if (filename.endsWith(".yml")) {
                     results.add(new JsonSchema(schemaContent));
-                }
-                if (filePath.endsWith(".proto")) {
+                } else if (filename.endsWith(".proto")) {
                     results.add(new ProtobufSchema(schemaContent));
+                } else {
+                    throw new UnsupportedOperationException("Unsupported schema file: " + filePath);
```

Previously, code was silently failing on unknown file type, which seemed wrong...